### PR TITLE
docs: Updated getting-started guide to add logout of helm registry

### LINKED ISSUE #3441: 403 error when trying to download helm chart
### Version

__Karpenter Version:__ v0.25.0

__Kubernetes Version:__ v1.21.0

### Expected Behavior

Able to download helm chart

### Actual Behavior

Receive a 403 error when trying to download helm chart (same for older versions too)

`Error: pulling from host public.ecr.aws failed with status code [manifests v0.25.0]: 403 Forbidden`

### Steps to Reproduce the Problem

`helm template karpenter oci://public.ecr.aws/karpenter/karpenter --version v0.25.0 --namespace karpenter \
    --set settings.aws.defaultInstanceProfile=KarpenterInstanceProfile \
    --set settings.aws.clusterEndpoint="" \
    --set settings.aws.clusterName=game-server-cluster \
    --set serviceAccount.annotations." \
    --version v0.25.0 > karpenter.yaml`

### Resource Specs and Logs

n/a

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by askulkarni2**
Fixes #3441

**Description**

Following the getting started guide on v0.27.3, got the following error even after `docker logout public.ecr.aws`.

```sh
docker logout public.ecr.aws

helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace karpenter --create-namespace \
  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
  --set settings.aws.clusterName=${CLUSTER_NAME} \
  --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
  --set settings.aws.interruptionQueueName=${CLUSTER_NAME} \
  --set controller.resources.requests.cpu=1 \
  --set controller.resources.requests.memory=1Gi \
  --set controller.resources.limits.cpu=1 \
  --set controller.resources.limits.memory=1Gi \
  --wait
Release "karpenter" does not exist. Installing it now.
Error: unexpected status from HEAD request to https://public.ecr.aws/v2/karpenter/karpenter/manifests/v0.27.3: 403 Forbidden
```

**How was this change tested?**

Followed https://github.com/aws/karpenter/issues/3441#issuecomment-1462355972 and that worked.

```sh
helm registry logout public.ecr.aws

helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace karpenter --create-namespace \
  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
  --set settings.aws.clusterName=${CLUSTER_NAME} \
  --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
  --set settings.aws.interruptionQueueName=${CLUSTER_NAME} \
  --set controller.resources.requests.cpu=1 \
  --set controller.resources.requests.memory=1Gi \
  --set controller.resources.limits.cpu=1 \
  --set controller.resources.limits.memory=1Gi \
  --wait
Release "karpenter" does not exist. Installing it now.
Pulled: public.ecr.aws/karpenter/karpenter:v0.27.3
Digest: sha256:81b08efedb60ad1323dcad9b93fb9d95153635f3cacef85f6d1aa7a86cd0d0ef
NAME: karpenter
LAST DEPLOYED: Thu May 11 13:41:48 2023
NAMESPACE: karpenter
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

**Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: #3441
- [ ] No

**Release Note**

```release-note
NONE
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/getting-started/getting-started-with-karpenter/scripts/step08-apply-helm-chart.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Logout of docker to perform an unauthenticated pull against the public ECR
 docker logout public.ecr.aws
 
+# Logout of helm registry
```
Comment (jonathan-innis) [inline website/content/en/preview/getting-started/getting-started-with-karpenter/scripts/step08-apply-helm-chart.sh:4]:
```suggestion
# Logout of helm registry to perform an authenticated pull against the public ECR
```

**Review by jonathan-innis — COMMENTED** [review]
Does `helm registry logout` work by itself or do you _absolutely_ need to perform both `docker logout` and `helm registry logout`

**Review by jonathan-innis — COMMENTED** [review]
Does `helm registry logout` work by itself or do you _absolutely_ need to perform both `docker logout` and `helm registry logout`

**Comment by askulkarni2** [comment]
I will test that and get back.

**Comment by askulkarni2** [comment]
@jonathan-innis looks like `helm registry logout` is sufficient.

```sh
» aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws

Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded

» helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace karpenter --create-namespace \
  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \ 
  --set settings.aws.clusterName=${CLUSTER_NAME} \
  --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
  --set settings.aws.interruptionQueueName=${CLUSTER_NAME} \
  --set controller.resources.requests.cpu=1 \
  --set controller.resources.requests.memory=1Gi \
  --set controller.resources.limits.cpu=1 \
  --set controller.resources.limits.memory=1Gi \
  --wait

Release "karpenter" does not exist. Installing it now.
Error: unexpected status from HEAD request to https://public.ecr.aws/v2/karpenter/karpenter/manifests/v0.27.3: 403 Forbidden

» helm registry logout public.ecr.aws
Removing login credentials for public.ecr.aws

» helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace karpenter --create-namespace \
  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \ 
  --set settings.aws.clusterName=${CLUSTER_NAME} \
  --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
  --set settings.aws.interruptionQueueName=${CLUSTER_NAME} \
  --set controller.resources.requests.cpu=1 \
  --set controller.resources.requests.memory=1Gi \
  --set controller.resources.limits.cpu=1 \
  --set controller.resources.limits.memory=1Gi \
  --wait

Release "karpenter" does not exist. Installing it now.
Pulled: public.ecr.aws/karpenter/karpenter:v0.27.3
Digest: sha256:81b08efedb60ad1323dcad9b93fb9d95153635f3cacef85f6d1aa7a86cd0d0ef
NAME: karpenter
LAST DEPLOYED: Fri May 12 09:55:49 2023
NAMESPACE: karpenter
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

**Comment by jonathan-innis** [comment]
@njtran Can you take a look at this one?

**Review by njtran — APPROVED** [review]

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**